ipygany/ipygany.py

```diff
@@ -651,9 +651,22 @@ class IsoColor(Effect):
     type = Enum(['linear', 'log'], default_value='linear').tag(sync=True)
 
     def __init__(self, parent, **kwargs):
+        colormap = kwargs.pop('colormap', None)
         super().__init__(parent, **kwargs)
         self.range = (self.min, self.max)
 
+        # set colormap
+        if isinstance(colormap, str):
+            if colormap not in colormaps:
+                allowed = ', '.join([f"'{clmp}'" for clmp in colormaps.keys()])
+                raise ValueError('``cmap`` "{cmap} is not supported by ``ipygany``\n'
+                                 'Pick from one of the following:\n' + allowed)
+            self.colormap = colormaps[colormap]
+        elif isinstance(colormap, int):
+            self.colormap = colormap
+        elif colormap is not None:
+            raise TypeError(f'Invalid colormap type {type(colormap)}')
```

Comment:
Thanks a lot for this change!

Instead of having this code in the `__init__`, it might be better to use a custom Traitlets validator for this colormap property (see [traitlets validation](https://traitlets.readthedocs.io/en/stable/using_traitlets.html#validation-and-coercion)).

Something like:
```python
@validate('colormap')
def _valid_colormap(self, proposal):
    colormap = proposal['value']
    # Put your validation code here, and raise a TraitError if it's not valid
    # Then return the value is it's an integer (trailets will validate that it's part of colormaps)
```

This way, it's not only the constructor keyword argument that supports a string, but you can also set the property like this:
```python
colored_mesh.colormap = 'Turbo'
colored_mesh.colormap = colormaps.Viridis
```

Reply:
I agree that validating traits with `@validate` is much cleaner, but I'm running into an issue where validate isn't accepting different types.  It seems that `validate` only works if you're attempting to input the same type.  Do you know of a workaround for this?  Otherwise, we implement our own handing of the trait.

Regardless, we should probably move this outside of the `__init__` to clean up the constructor.

Reply:
Too bad it doesn't work like that. Then maybe defining `colormap` as being a `traitlets.Any` and handling the validation on our own is the best we can do here?

Reply:
I think so.  This seems to be the only way we can allow non-integer colormaps.  I'll update it and you can let me know what you think.

Reply:
Thanks a lot!